controllers/contact-controller.js

```diff
@@ -67,6 +67,80 @@ function isEmptySearchParams(body) {
 
 const orUndefined = value => value || undefined;
 
+const queryOnName = operator => singleInput => firstName => lastName =>
```

Comment:
Out of curiosity, what is the reason for currying the parameters instead of listing them?